templates/manage/email_accounts.tpl.html

```diff
@@ -91,6 +79,18 @@
       toggleFolderField();
       $('#email_account_form').submit(validateForm);
       $('.select_all').click(function() { Eventum.toggleCheckAll('items[]'); });
+      $('#testSettings').click(function(event) {
+           event.preventDefault();
+           $('#full_width_messages').removeClass().html('Please wait.. testing connection.').addClass('info');
+           var serializedData = $('#email_account_form').serialize();
+           $.ajax({
+               type: "POST",
+               url: "check_email_settings.php",
```

Comment:
I agree a shared JS function would probably be better.  For authentication,
I have always used the same authentication on the ajax PHP files as
others.  The page still has access to the session, do we need to do
anything differently than for example list.php?

I see what you mean in the ajax/ sub directory.  Any reason we couldn't add
`Auth::checkAuthentication();` to the bottom of init.php and avoid calling
it in every other file? We would have to skip authentication if init.php
was included from a cli call (wouldn't we?).

Let me know what you and @balsdorf think and I'll rework this example.

Craig.

On Fri, Nov 6, 2015 at 4:15 PM, Elan Ruusamäe <notifications@github.com>
wrote:

> In templates/manage/email_accounts.tpl.html
> <https://github.com/eventum/eventum/pull/96#discussion_r44188285>:
>
> > @@ -91,6 +79,18 @@
> >        toggleFolderField();
> >        $('#email_account_form').submit(validateForm);
> >        $('.select_all').click(function() { Eventum.toggleCheckAll('items[]'); });
> > +      $('#testSettings').click(function(event) {
> > +           event.preventDefault();
> > +           $('#full_width_messages').removeClass().html('Please wait.. testing connection.').addClass('info');
> > +           var serializedData = $('#email_account_form').serialize();
> > +           $.ajax({
> > +               type: "POST",
> > +               url: "check_email_settings.php",
>
> i think we should create common entry point for ajax calls. there are too
> many of them, and each one does things by it's own, not sure which is even
> correct one. i.e how to properly check for authenticated user and role.
>
> see here:
> https://github.com/eventum/eventum/tree/v3.0.5/htdocs/ajax
>
> they all do the authenticity check differently.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/eventum/eventum/pull/96/files#r44188285>.
>